sentry-android-timber/src/main/java/io/sentry/android/timber/SentryTimberTree.kt

```diff
@@ -0,0 +1,74 @@
+package io.sentry.android.timber
+
+import android.util.Log
+import io.sentry.core.IHub
+import io.sentry.core.SentryEvent
+import io.sentry.core.SentryLevel
+import io.sentry.core.protocol.Message
+import timber.log.Timber
+
+/**
+ * Sentry Timber tree which is responsible to capture events via Timber
+ */
+class SentryTimberTree(private val hub: IHub, private val minLevel: SentryLevel) : Timber.Tree() {
+
+    /**
+     * Compares the minLevel given by the user and the level given by the event
+     * to capture an event or not.
+     */
+    override fun isLoggable(tag: String?, priority: Int): Boolean {
+        val sentryLevel = getSentryLevel(priority)
+
+        // do not log if it's lower than min. required level.
+        return sentryLevel.ordinal >= minLevel.ordinal
+    }
+
+    /**
+     * Captures a Sentry Event if the min. level is equal or higher than the min. required level.
+     */
+    override fun log(priority: Int, tag: String?, message: String, throwable: Throwable?) {
+        val sentryEvent = SentryEvent()
+        sentryEvent.level = getSentryLevel(priority)
+
+        // if there's no throwable, should it be a breadcrumb then?
```

Comment:
@bruno-garcia wdyt?

Reply:
In other SDKs there were 2 log levels you could configure the library with:


Both checks run always, so potentially a Log call will result in both Event and Breadcrumb: 
https://github.com/getsentry/sentry-dotnet/blob/17c5dabac328181c6f700f5afb07d301cc0183d6/src/Sentry.Extensions.Logging/SentryLogger.cs#L57-L99

One to configure min level for event:
https://github.com/getsentry/sentry-dotnet/blob/17c5dabac328181c6f700f5afb07d301cc0183d6/src/Sentry.Extensions.Logging/SentryLogger.cs#L125

One to configure min level for breadcrumb: https://github.com/getsentry/sentry-dotnet/blob/17c5dabac328181c6f700f5afb07d301cc0183d6/src/Sentry.Extensions.Logging/SentryLogger.cs#L144

You can check/send event first, and run the check to add breadcrumb later. The advantage is that a second event will include the previous error in its breadcrumb list.

Reply:
good idea, just added it, thx